tests/core/test_channels.py

```diff
@@ -369,6 +369,29 @@ def test_socketio_channel():
     assert routes_list["handle_request"].startswith("/socket.io")
 
 
+def test_socketio_channel_metadata():
```

Comment:
This test is a great start, however we need to check that the metadata given to `handle_message` is passed on correctly to the `UserMessage`, is that feasible to add?

Reply:
I agree. I was looking for an existing test in the REST & Socket channels that do a test that a message was received (figured I would copy that test and add a check for the metadata). There must be existing tests for this but I couldn't find one.

Reply:
Test suite might not be as comprehensive as we wish 😅 
However there seem to be some ideas in other channel test modules, for example you could create a Sanic app instance using an empty Agent for speedy test execution or some other agent fixtures from`conftest.py`, and make assertions to the response from `app.test_client.post(webhook, headers, data)`?
You can check out these unit tests:
- [twilio](https://github.com/RasaHQ/rasa/blob/593d5265769a5d03a64b3102c7371a341cf7d8a2/tests/core/channels/test_twilio_voice.py#L226)
- [telegram](https://github.com/RasaHQ/rasa/blob/593d5265769a5d03a64b3102c7371a341cf7d8a2/tests/core/channels/test_telegram.py#L25)

Reply:
I've added a [test_socketio.py](https://github.com/rgstephens/rasa/blob/eb87042fbdebe4a35232a853f8dd7262f4cc7afe/tests/core/channels/test_socketio.py#L44) but I'm not very confident about what this is doing. Would you take a look?

Reply:
The test you added seems to be testing the output channel?
I made the below unit test to make sure a message with metadata gets processed correctly, however stumbled across an error as described in this [ticket](https://github.com/RasaHQ/rasa/issues/11084):

```
async def test_socketio_metadata(default_agent):
    import rasa.server
    app = rasa.server.create_app(agent=default_agent)

    from rasa.core.channels.socketio import SocketIOInput

    input_channel = SocketIOInput(
        # event name for messages sent from the user
        user_message_evt="user_uttered",
        # event name for messages sent from the bot
        bot_message_evt="bot_uttered",
        # socket.io namespace to use for the messages
        namespace=None,
    )
    rasa.core.channels.channel.register([input_channel], app, "/webhooks/")

    client = app.asgi_client

    body = {"metadata": "test", "message": "Hello!", "session_id": "test_id"}

    _, response = await client.post("/socket.io", data=body)
    print(response.body)
    assert response.status == HTTPStatus.OK
```
We'll look into this ticket soon.

Reply:
As discovered by @znat, the endpoint is missing a query parameter "/socket.io/\?EIO\=4" which is why the response status was 400. However the test still fails with status 404 URL not found even after this edit. I've run out of ideas here: unless @znat can spot what's the issue, I would say we can skip adding a unit test as long as @rgstephens can confirm that the PR works fine against a manual test implementing socketio channel.

Reply:
I removed the test and fixed an issue when the message is null